analyzers/tests/SonarAnalyzer.UnitTest/MetadataReferences/NuGetMetadataFactory.cs

```diff
@@ -83,47 +83,26 @@ public static IEnumerable<MetadataReference> CreateNETStandard21()
                .ToImmutableArray();
         }
 
-        private static IEnumerable<MetadataReference> Create(Package package, string[] allowedTargetFrameworks)
+        private static IEnumerable<MetadataReference> Create(Package package, string[] allowedDirectories)
```

Comment:
please add a comment documenting the assumption that `allowedDirectiories` should be sorted - I believe `dllsPerDirectory[allowedDirectories.First(x => dllsPerDirectory.ContainsKey(x))];` relies on that

and could also name the parameter "sortedAllowedTargetFrameworkDirectories", explaining in the method comment that the directories are going to be verified one after another in the order they are given in

Reply:
Given the DLL situation in the last commit, it's not really a target framework in all cases. `allowedDirectory` is more precise name.

I'll add the sorted comment.

Reply:
Thinking back on the to topic, the `target framework` component was actually a source of confusion in the deobfuscation phase.

Reply:
We can consider renaming `Create(.., targetFramework)` argument to something else, like `dllDirectory` and update the comment accordingly.